Ginger/Ginger/ApplicationModelsLib/POMModels/POMWizardLib/LearnWizard/AIGeneratedPreviewWizardPage.xaml.cs

```diff
@@ -0,0 +1,242 @@
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+
+namespace Ginger.ApplicationModelsLib.POMModels.AddEditPOMWizardLib
+{
+    /// <summary>
+    /// Interaction logic for POMScreenshotHTMLViewPage.xaml
+    /// </summary>
+    public partial class AIGeneratedPreviewWizardPage : Page, IWizardPage
+    {
+        private AddPOMFromScreenshotWizard mWizard;
+
+
+        ApiSettings ApiSettings { get; set; }
+        public AIGeneratedPreviewWizardPage()
+        {
+            InitializeComponent();
+        }
+
+        public void WizardEvent(WizardEventArgs WizardEventArgs)
+        {
+            switch (WizardEventArgs.EventType)
+            {
+                case EventType.Init:
+                    mWizard = (AddPOMFromScreenshotWizard)WizardEventArgs.Wizard;
+                    ApiSettings = LoadApiSettings();
+                    if (ApiSettings == null)
+                    {
+                        Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "OpenAI settings could not be loaded.");
+                        WizardEventArgs.Wizard.Cancel();   // or disable page
+                        return;
+                    }
+                    break;
+                case EventType.Active:
+                    if(string.IsNullOrEmpty(ApiSettings.ApiKey) || ApiSettings.ApiKey.Equals("YouuAPiKey"))
+                    {
+                        Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "OpenAI setting are not valid.");
+                        WizardEventArgs.Wizard.Cancel();   // or disable page
+                        break;
+                    }
+                    GenerateHtmlAsync();
+                    break;
+                case EventType.LeavingForNextPage:
+                    break;
+            }
+        }
+
+        public ApiSettings LoadApiSettings()
+        {
+            try
+            {
+                // Get the directory of the current class
+                string directory = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+
+                // Combine the directory with the file name
+                string filePath = Path.Combine(directory, "OpenAIappsetting.json");
+
+                string json = File.ReadAllText(filePath);
+
+
+                var root = JsonConvert.DeserializeObject<Root>(json);
+
+                return root.ApiSettings;
+
+            }
+
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to load OpenAI API settings", ex);
+                return null;
+            }
+        }
+
+        private async void GenerateHtmlAsync()
+        {
+            if(!string.IsNullOrEmpty(mWizard.HtmlFilePath))
```

Comment:
Add try catch and put mWizard.ProcessEnded(); in finally

Reply:
done